lib/IRGen/GenProto.cpp

```diff
@@ -511,7 +511,14 @@ void EmitPolymorphicParameters::bindExtraSource(const MetadataSource &source,
       auto selfTy = FnType->getSelfInstanceType();
       CanType argTy = getTypeInContext(selfTy);
       setTypeMetadataName(IGF.IGM, metadata, argTy);
-      IGF.bindLocalTypeDataFromTypeMetadata(argTy, IsExact, metadata);
+      auto *CD = selfTy.getClassOrBoundGenericClass();
+      // If it is a class and it is not final, the metadata is not exact
+      // for argTy, because it may be a metadata for a derived class.
+      // In all other cases, the metadata is exact.
+      // TODO: If it can be proved that the class has no subclasses 
+      // (e.g. in WMO mode), then the metadata can be considered exact.
```

Comment:
I would suggest this comment:

The self metadata here corresponds to the conforming type.  For an inheritable conformance, that may be a subclass of the static type, and so the self metadata will be inexact.  Currently, all conformances are inheritable.